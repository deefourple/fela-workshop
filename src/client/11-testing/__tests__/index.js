import React from 'react';
import { Button } from '../index';
import { mount } from 'enzyme';
import { felaSnapshot, felaTestContext } from 'cf-style-provider';

test('button renders', () => {
  const snapshot = felaSnapshot(<Button>Hai</Button>);
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});

test('active button renders', () => {
  const snapshot = felaSnapshot(<Button active>Haiii</Button>);
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});

test('should call onClick function', () => {
  const handleClick = jest.fn();
  const wrapper = mount(
    felaTestContext(<Button onClick={handleClick}>Hit meeeee</Button>)
  );
  expect(handleClick.mock.calls.length).toBe(0);
  wrapper.find(Button).simulate('click');
  expect(handleClick.mock.calls.length).toBe(1);
});
