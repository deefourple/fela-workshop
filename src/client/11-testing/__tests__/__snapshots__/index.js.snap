// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`active button renders 1`] = `
<button
  className="a b"
  onClick={undefined}
>
  Haiii
</button>
`;

exports[`active button renders 2`] = `
"
.a {
  padding: 2rem
}

.b {
  background-color: #ccc
}
"
`;

exports[`button renders 1`] = `
<button
  className="a b"
  onClick={undefined}
>
  Hai
</button>
`;

exports[`button renders 2`] = `
"
.a {
  padding: 2rem
}

.b {
  background-color: black
}
"
`;
